### PR TITLE
Deleting a branch calls dialog of deleting its worktree (GLVSC-632)

### PR DESCRIPTION
**Pull request by sergeibbb**
# Description

Or instead, the Delete Worktree flow could just be triggered as an optional step in the Delete Branch flow.

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes include any required corresponding changes to the documentation (including CHANGELOG.md and README.md)
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

**Review by axosoft-ramint — COMMENTED** [review]
Just commenting on the user experience, but when I choose to delete a branch, I find it a bit stange that it jumps me into this screen:

image

There is no indication here that "this is a branch with a worktree, so you need to delete the worktree first". It just says "delete worktree". So it feels like I clicked the wrong command, even though I did not.

I think we can do a better job with messaging to make this less confusing. Either the title of the "delete worktree" step can be updated, or the two available options can say "Delete Worktree & Branch" and "Force Delete Worktree & Branch", to make it clear that we are deleting the worktree and then the branch.

File: src/commands/git/worktree.ts [hunk 1]
```diff
@@ -110,6 +110,7 @@ interface DeleteState {
 	repo: string | Repository;
 	uris: Uri[];
 	flags: DeleteFlags[];
+	doNotSuggestDeletingBranches?: boolean;
```
Comment (axosoft-ramint) [inline src/commands/git/worktree.ts:113]: Maybe we can just pass `branch` or `branches` as state when coming from "delete branch" and then we don't need to calculate them on the delete steps, and also can use their existence in state as a flag to restrict the options?

Comment (sergeibbb) [reply src/commands/git/worktree.ts:113]: @axosoft-ramint 

What do you think about this:
image

Comment (sergeibbb) [reply src/commands/git/worktree.ts:113]: @axosoft-ramint 

>  then we don't need to calculate them on the delete steps

It's not clear how we can use the passed branches on the "worktrees" step. We calculate branches only if "Delete Worktrees&Branches" is selected, but those are hidden.

Comment (axosoft-ramint) [reply src/commands/git/worktree.ts:113]: Fair enough - if the state isn't needed then a boolean should suffice 👍🏼

Comment (sergeibbb) [reply src/commands/git/worktree.ts:113]: @axosoft-ramint 

I'm thinking about renaming because it affects the view in a very specific case. "Deleting of the worktree of a selected branch". So, it can be called like this: `deletingOfSelectedBranches`, or moreover I'm thinking of replacing the field by a flag:

```
type DeleteFlags = '--force' | '--delete-branches' | '--deleting-of-selected-branches';
```

What do you think?

Comment (sergeibbb) [reply src/commands/git/worktree.ts]: @axosoft-ramint 

I've just pushed how it can be with a `--deleting-of-selected-branches` flag. Let me know if you prefers it being a boolean field.

Comment (sergeibbb) [reply src/commands/git/worktree.ts]: The UI:

image

image

Unfortunately, I need to go now. I can do any other updates tomorrow.

Comment (axosoft-ramint) [reply src/commands/git/worktree.ts]: @sergeibbb No preference here. Up to you. Doing it with a flag like that sounds fine. My main hope is that the messaging on the worktree step allows me to understand that we are deleting the worktree because it is necessary to delete the branch.

Comment (sergeibbb) [reply src/commands/git/worktree.ts]: @axosoft-ramint 

Ok. I'm leaving it with the flag. Feel free to suggest updates for the wording, so it would be more clearer for user. You are the native speaker, so, you know it better.

Comment (axosoft-ramint) [reply src/commands/git/worktree.ts]: @sergeibbb I think we can take out the word "corresponding" in the options since it becomes a bit wordy. We should also make sure the capitalization is consistent in the options.

What do you think about "Delete Worktree of Branch"/"Delete Worktrees of Branch" etc.?

Comment (axosoft-ramint) [reply src/commands/git/worktree.ts]: @sergeibbb I updated the wording on your branch. Thanks for working on this!

**Review by axosoft-ramint — APPROVED** [review]
LGTM

File: src/commands/git/worktree.ts (hunk 1 above)
Comment (sergeibbb) [reply src/commands/git/worktree.ts]: Great! Thank you!
